core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java

```diff
@@ -320,12 +321,20 @@ public void testCertAuthenticator()
     @Test
     public void testJwtAuthenticator()
             throws Exception
+    {
+        verifyJwtAuthenticator(Optional.empty());
+        verifyJwtAuthenticator(Optional.of("custom-principal"));
+    }
+
+    private void verifyJwtAuthenticator(Optional<String> principalField)
+            throws Exception
     {
         try (TestingTrinoServer server = TestingTrinoServer.builder()
                 .setProperties(ImmutableMap.<String, String>builder()
                         .putAll(SECURE_PROPERTIES)
                         .put("http-server.authentication.type", "jwt")
                         .put("http-server.authentication.jwt.key-file", HMAC_KEY)
+                        .put("http-server.authentication.jwt.principal-field", principalField.orElse("sub"))
```

Comment:
It might be better to only set this property if it is custom, so that the default is used otherwise, as that better reflects how it is used in practice.

Reply:
We have other tests to verify the default config values.  This makes the tests much easier to write because I can do it inline.